avocado/plugins/runner_nrunner.py

```diff
@@ -192,9 +192,9 @@ def _create_runtime_tasks_for_test(test_suite, runnable, no_digits,
         result = []
 
         # test related operations
-        # when creating variants, they need to have a copy of the runnable.
-        if len(test_suite.tests) == 1 and index > 1:
-            runnable = deepcopy(runnable)
+        # to avoid complex code determining if it is a variant or a single
```

Comment:
Maybe just add a note that, we currently have the variants coupled to the runnable, so it's not possible at this time to optmize too much.  Bonus points for an issue describing the issue.  Extra bonus points for having a link here :)